src/components/Weather.js

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from 'react';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import Cities from './Cities';
+import Details from './Details';
+import { getWeatherFromRandomCities } from '../redux/weather/weatherSlice';
+
+const Weather = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getWeatherFromRandomCities());
+  }, []);
+
+  return (
+    <BrowserRouter>
+      {/* Your application here */}
+      <Routes>
+        <Route path="/" element={<Cities />} />
+        <Route path="/details" element={<Details />} />
+      </Routes>
+    </BrowserRouter>
```

Comment:
- [ ] As mentioned previously, kindly make sure you are using proper `Routing` on your `details` to make sure the user gets the details for each specific `city` based on it `id` or `name.` i.e you may say;
```jsx
<Route path="/details/:id(name)" element={<Details />} />
// And make sure the use navigate is used correctly.
```
You may consider check [here](https://www.geeksforgeeks.org/reactjs-usenavigate-hook/) to see how the `UseNvagivate` hook works